Sources/SourceKitLSP/Swift/DocumentSymbols.swift

```diff
@@ -0,0 +1,241 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import SwiftSyntax
+import LSPLogging
+
+extension SwiftLanguageServer {
+  public func documentSymbol(_ req: DocumentSymbolRequest) async throws -> DocumentSymbolResponse? {
+    guard let snapshot = self.documentManager.latestSnapshot(req.textDocument.uri) else {
+      logger.error("failed to find snapshot for url \(req.textDocument.uri.forLogging)")
```

Comment:
We have this in so many places. Is requesting a snapshot and not finding it ever *not* an error? ie. should we just always log?

Reply:
Good idea. Here we go: https://github.com/apple/sourcekit-lsp/pull/943